### PR TITLE
added return value of bash module.

### PR DESCRIPTION
**Pull request by huzai9527**
the commond which been exec successfully will return 0,otherwhise will return errcode.
image

### CODE REVIEW

File: kern/bash_kern.c [hunk 1]
```diff
@@ -3,13 +3,20 @@
...
 struct {
     __uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
 } events SEC(".maps");
 
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __type(key, u32);
+    __type(value, struct event);
+    __uint(max_entries, 1024);
+}events_t SEC(".maps");
```
Comment (yihong0618) [inline kern/bash_kern.c:19]: maybe need one space. } events_t

File: kern/bash_kern.c [hunk 2]
```diff
@@ -25,14 +32,32 @@ int uretprobe_bash_readline(struct pt_regs *ctx) {
     }
 #endif
 
-    struct event event;
-    //    bpf_printk("!! uretprobe_bash_readline pid:%d",target_pid );
+    struct event event={};
```
Comment (yihong0618) [inline kern/bash_kern.c:35]: struct event event = {};

**Review by yihong0618 — COMMENTED** [review]
seems need some code format to follow this `repo` c code style.

**Comment by cfc4n** [comment]
and merge all commits into one commit with command
```shell
git reset --soft HEAD~3
git commit
git push -f
```

and This PR will autorefresh

File: user/probe_bash.go [hunk 3]
```diff
@@ -104,6 +104,7 @@ func (this *MBashProbe) setupManagers() {
 	}
 
 	this.logger.Printf("HOOK binrayPath:%s, FunctionName:readline\n", binaryPath)
+	this.logger.Printf("HOOK binrayPath:%s, FunctionName:execute_command\n", binaryPath)
```
Comment (cfc4n) [inline user/probe_bash.go:107]: Is it repeated here?   Line 106 and 107

File: user/probe_bash.go [hunk 4]
```diff
@@ -114,6 +115,12 @@ func (this *MBashProbe) setupManagers() {
 				//UprobeOffset: 0x8232, 	//若找不到 readline 函数，则使用offset便宜地址方式。
 				BinaryPath: binaryPath, // 可能是 /bin/bash 也可能是 readline.so的真实地址
 			},
+			{
+				Section:          "uretprobe/bash_retval",
+				EbpfFuncName:     "uretprobe_bash_retval",
+				AttachToFuncName: "execute_command",
```
Comment (cfc4n) [inline user/probe_bash.go:121]: Can you give some source code about bash return?

**Review by cfc4n — COMMENTED** [review]
please fix them ,and run `make format` to format c code before `git commit`

File: user/probe_bash.go (hunk 3 above)
Comment (huzai9527) [reply user/probe_bash.go:107]: 107 is another hook, `execute_command`

**Comment by huzai9527** [comment]
I have run the command up, is that works? I want to add falgs on bash command which will be used like this `sudo ./ecapture bash -err 100`, this command will show user which command don't have permisson to run.

File: user/probe_bash.go (hunk 4 above)
Comment (huzai9527) [reply user/probe_bash.go:121]: the bash workflow like this
```
 main()
        |
   reader_loop()       解析
        |--------------------------->read_command()-->parse_command()-->yyparse()-->yylex()-->read_token()-->read_token_word()
        |                                 |                               |                       |                 |
 execute_command() <-------------- current_command <--------------- global_command <------------token------------word
        |
execute_command_internal()
        |
 execute_xxxx_command()
        |
execute_simple_command()
        |
        |--->expand_words()-->expand_word_list_internal()
        |                                                                  子进程
        |------------------------------------->execute_disk_command()------------->shell_execve()-->execve()                
        |                  磁盘命令                       |                |                       |
        |函数及内置命令                              make_child()          |                       |FAILED
        |                                                |                |                       |
execute_builtin_or_function()                          fork()----------->pid                      ->execute_shell_script()
                                                                          |
                                                                          --------->return(result)
                                                                            父进程
```

**Comment by cfc4n** [comment]
as code :

https://github.com/ehids/ecapture/blob/13ea96405cfbcac842cdb2367d6e466942766a35/cli/cmd/bash.go#L29-L30

**Comment by huzai9527** [comment]
copy that, Do you have any other issues with this PR?

**Comment by huzai9527** [comment]
copy that, Do you have any other issues with this PR?

**Comment by huzai9527** [comment]
I have implemented  the command like `sudo ./ecapture bash --err 100`  and it's result like this
- with err flag
image

- no err flag
image

**Comment by cfc4n** [comment]
please merge all commits into a commit.  ref: https://www.internalpointers.com/post/squash-commits-into-one-git

**Comment by huzai9527** [comment]
sorry about these commits, I am not familiar with github's instruction, I am trying to merge all commits now.

**Comment by cfc4n** [comment]
加下  https://raw.githubusercontent.com/ehids/ecapture/master/images/wechat_gzhh.png 微信公众号，里面回复“微信群”，会给发给你群二维码，我们群里沟通吧。

**Comment by huzai9527** [comment]
群二维码过期啦

**Review by cfc4n — COMMENTED** [review]
LGTM

**Pull request merged**